cc/layers/layer.cc

```diff
@@ -73,6 +73,16 @@ static_assert(sizeof(Layer) == sizeof(SameSizeAsLayer),
               "Layer should stay small");
 
 base::AtomicSequenceNumber g_next_layer_id;
+size_t next_layer_lock = recordreplay::CreateOrderedLock("Layer::NextID");
+
+// https://linear.app/replay/issue/RUN-1229
+inline int NextLayerId() {
+  recordreplay::OrderedLock(next_layer_lock);
```

Comment:
This is a good find an honestly a likely root cause of the later mismatch.  @bhackett1024 thoughts?

Reply:
It's possible, generally I'd prefer it though if we made sure a thread ordering issue was definitely a problem before adding more ordered locks.  It's not clear from the chromium commit whether layers can be constructed on multiple threads in a normal render process.  In this case we could add an assertion in the layer constructor on its ID and see if that gets tripped.  This isn't a big deal though and it's fine to land the patch as is.